gog/graph.go

```diff
@@ -158,6 +158,15 @@ func (v *astVisitor) Visit(node ast.Node) (w ast.Visitor) {
 			return nil
 		}
 
+	case *ast.SelectorExpr:
+		if sel := v.typesInfo.Selections[n]; sel != nil {
+			t := sel.Recv()
+			if ptr, ok := t.(*types.Pointer); ok {
+				t = ptr.Elem()
```

Comment:
If you have the expression `a.b`, then `a` can be either a struct or a pointer to a struct. In the latter case, `b` needs to be registered as a field of the struct, not of the pointer.